security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -407,22 +407,26 @@ func addConn(k cache.ConnKey, conn *sdsConnection) {
 func pushSDS(con *sdsConnection) error {
 	response, err := sdsDiscoveryResponse(con.secret, con.conID)
 	if err != nil {
-		sdsServiceLog.Errorf("SDS: Failed to construct response %v", err)
+		sdsServiceLog.Errorf("Failed to construct response for SDS resource %q: %v", con.ResourceName, err)
```

Comment:
here may trigger racing condition when reading con.resourcename, see rlock in this func. 
probably using rlock to read resourcename, then use it in debug logs.

Reply:
Good point. Revised. Please take a look.